WearTilesKotlin/app/src/main/java/com/example/wear/tiles/messaging/MessagingTileRenderer.kt

```diff
@@ -51,223 +31,34 @@ class MessagingTileRenderer(context: Context) :
         state: MessagingTileState,
         deviceParameters: DeviceParametersBuilders.DeviceParameters
     ): LayoutElement {
-        return tileLayout(state, deviceParameters)
+        return messagingTileLayout(state, context, deviceParameters)
     }
 
+    /**
+     * If we want to display an image in our layout, we have to add the image resource to the
+     * [Resources.Builder].
+     */
     override fun Resources.Builder.produceRequestedResources(
```

Comment:
TIOLI: SingleTileLayoutRenderer is hiding the complexity of versions and effect on resources. I wonder whether we should expose this to override in SingleTileLayoutRenderer, or if not, put a quick doc comment here to explain that images will be cached permanently, and ids should have a timestamp if we want them to evolve, or we should change the id to flush all...

Reply:
Another thing that SingleTileLayoutRenderer hides is freshnessIntervalMillis, is it worth overriding in this class to document, even if we set it to 0, and say that updates are coming from externally pushed? Or set it to 5 minutes?

Reply:
For versioning, added a note here to revisit this https://github.com/android/wear-os-samples/issues/496

re: the freshness thing - in this sample, I'm not sure that highlighting it will make anything clearer because we never update the tile externally and setting to a non-zero value (>0 means system will trigger refreshes) doesn't help either because the data never changes.